vector/src/main/java/im/vector/app/features/media/VectorAttachmentViewerViewModel.kt

```diff
@@ -0,0 +1,74 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.media
+
+import com.airbnb.mvrx.MavericksViewModelFactory
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import im.vector.app.core.di.MavericksAssistedViewModelFactory
+import im.vector.app.core.di.hiltMavericksViewModelFactory
+import im.vector.app.core.platform.VectorDummyViewState
+import im.vector.app.core.platform.VectorViewModel
+import im.vector.app.features.media.domain.usecase.DownloadMediaUseCase
+import kotlinx.coroutines.launch
+
+class VectorAttachmentViewerViewModel @AssistedInject constructor(
+        @Assisted initialState: VectorDummyViewState,
+        private val downloadMediaUseCase: DownloadMediaUseCase
+) : VectorViewModel<VectorDummyViewState, VectorAttachmentViewerAction, VectorAttachmentViewerViewEvents>(initialState) {
+
+    /* ==========================================================================================
+     * Factory
+     * ========================================================================================== */
+
+    @AssistedFactory
+    interface Factory : MavericksAssistedViewModelFactory<VectorAttachmentViewerViewModel, VectorDummyViewState> {
+        override fun create(initialState: VectorDummyViewState): VectorAttachmentViewerViewModel
+    }
+
+    companion object : MavericksViewModelFactory<VectorAttachmentViewerViewModel, VectorDummyViewState> by hiltMavericksViewModelFactory()
+
+    /* ==========================================================================================
+     * Public Api
+     * ========================================================================================== */
+
+    var pendingAction: VectorAttachmentViewerAction? = null
+
+    /* ==========================================================================================
+     * Specialization
+     * ========================================================================================== */
+
+    override fun handle(action: VectorAttachmentViewerAction) {
+        when (action) {
+            is VectorAttachmentViewerAction.DownloadMedia -> handleDownloadAction(action)
+        }
+    }
+
+    /* ==========================================================================================
+     * Private methods
+     * ========================================================================================== */
+
+    private fun handleDownloadAction(action: VectorAttachmentViewerAction.DownloadMedia) {
+        viewModelScope.launch {
```

Comment:
Should use `session.coroutineScope` to avoid binding the download to the lifecycle of the `ViewModel`.